bound/kt/src/main/kotlin/web5/sdk/dids/BearerDid.kt

```diff
@@ -0,0 +1,78 @@
+package web5.sdk.dids
+
+import web5.sdk.crypto.signers.Signer
+import web5.sdk.crypto.keys.KeyManager
+import web5.sdk.crypto.signers.OuterSigner
+
+import web5.sdk.rust.BearerDid as RustCoreBearerDid
+
+import web5.sdk.rust.didJwkResolve as rustCoreDidJwkResolve
+import web5.sdk.rust.didWebResolve as rustCoreDidWebResolve
+import web5.sdk.rust.didDhtResolve as rustCoreDidDhtResolve
+
+/**
+ * Represents a Decentralized Identifier (DID) along with its DID document, key manager, metadata,
+ * and convenience functions.
+ *
+ * @property did The DID associated with this instance.
+ * @property document The DID document associated with this instance.
+ */
+class BearerDid {
+    val did: Did
+    val document: Document
+    val keyManager: KeyManager
+
+    private val rustCoreBearerDid: RustCoreBearerDid
+
+    /**
+     * Constructs a BearerDid instance using a DID URI and a key manager.
+     *
+     * @param uri The DID URI.
+     * @param keyManager The key manager to handle keys.
+     */
+    constructor(uri: String, keyManager: KeyManager) {
+        this.rustCoreBearerDid = RustCoreBearerDid(uri, keyManager.getRustCoreKeyManager())
+
+        this.did = this.rustCoreBearerDid.getData().did
+        this.document = this.rustCoreBearerDid.getData().document
+        this.keyManager = keyManager
+    }
+
+    /**
+     * Returns a signer for the DID.
+     *
+     * @return Signer The signer for the DID.
+     */
+    fun getSigner(): Signer {
+        val keyId = this.document.verificationMethod.first().id
+        val innerSigner = this.rustCoreBearerDid.getSigner(keyId)
+        return OuterSigner(innerSigner)
+    }
+
+    companion object {
+        /**
+         * Resolves a DID URI to a ResolutionResult.
+         *
+         * @param uri The DID URI to resolve.
+         * @return ResolutionResult The result of the DID resolution.
+         */
+        @JvmStatic
+        suspend fun resolve(uri: String): ResolutionResult {
```

Comment:
What's your thinking with this function? This isn't in the APID, is it?

Reply:
right, I think got influenced by current impl in web5-kt.

That does beg the question though, how would we resovle a bearer did:

It would have to be like:

```
DidDht.resolve(bearerDid.did.uri)
```

But how would you know the type, (the bearer did could be JWK..)

Reply:
```
CLASS BearerDid
  PUBLIC DATA did: Did
  PUBLIC DATA document: Document
  CONSTRUCTOR(uri: string, key_manager: KeyManager)
  METHOD get_signer(): Signer
```

Right genius, its already resolved basically and we have document right there :duhh: I'll update it